synapse/handlers/room_member.py

```diff
@@ -697,6 +699,37 @@ def _lookup_3pid(self, id_server, medium, address):
             raise SynapseError(
                 403, "Looking up third-party identifiers is denied from this server"
             )
+
+        # Check what hashing details are supported by this identity server
+        try:
+            hash_details = yield self.simple_http_client.get_json(
+                "%s%s/_matrix/identity/v2/hash_details" % (id_server_scheme, id_server)
+            )
+        except (HttpResponseException, ValueError) as e:
```

Comment:
A `ValueError` is returned by `SimpleHttpClient.get_json` if the response was not JSON, which may be the case on a 404 error? I'm not sure if it prioritizes a 404 error code over receiving non-JSON from the server, but even a non-JSON 200 error is unusable.

Though perhaps it could be argued that we should just fail the request in that case versus falling back to v1.

Reply:
Ugh, that's annoying.

Reply:
I've added comments explaining the reason for catching these exception types.